scripts/sct_dmri_moco.py

```diff
@@ -84,8 +84,7 @@ def get_parser():
                                   "poly [int]: Degree of polynomial function used for regularization along Z. For no regularization set to 0. Default=" + param_default.poly + ".\n"
                                                 "smooth [mm]: Smoothing kernel. Default=" + param_default.smooth + ".\n"
                                                   "metric {MI, MeanSquares, CC}: Metric used for registration. Default=" + param_default.metric + ".\n"
-                                                  "gradStep [float]: Searching step used by registration algorithm. The higher the more deformation allowed. Default=" + param_default.gradStep + ".\n"
-                                                    "sample [0-1]: Sampling rate used for registration metric. Default=" + param_default.sampling + ".\n",
```

Comment:
@kousu should we really remove this parameter? Are there no arguments for allowing users to set the sampling density? From a testing perspective it makes sense to use full, but maybe users would want to set this if they're time/resource constrained.

Reply:
That's a big open question to me here. From the thread https://github.com/ANTsX/ANTs/issues/976 we learned that dense sampling and single-threading was the only way to get reproducible results. @jcohenadad suggested to me that our users would also want perfectly reliable results, so leaning on accuracy over speed this PR hardcodes both those in.

I could go either way on the issue. I could see it being equally good to relax the numerical tightness in the tests -- as the [author says](https://github.com/ANTsX/ANTsR/issues/210#issuecomment-377511054).

> If the ​population statistical ​ outputs are sensitive to tiny variations in rotation parameters, then there are other issues one should worry about.

Reply:
I would still want the default parameter to provide reproducible results (ie: dense with single thread), however it would be indeed nice to keep the option to modify sampling. Especially if people have slow computers and big data.

Reply:
I put `sampling` back. I just added a new `sampling=None` option.

I didn't *typecheck* it though. If a user types `sampling=none` it will fail. My reasoning is that we already don't typecheck that argument, and passing an incorrect value fails deep, when passed to antsRegistration:

```
$ spinalcordtoolbox/bin/isct_antsSliceRegularizedRegistration --polydegree 2 --transform 'Translation[1]' --metric 'MI[target.nii.gz,/tmp/sct-20200517024148.039279-moco-7p9n_nbk/b0_T0000.nii,1,16,Regular,9]' --iterations 10 --shrinkFactors 1 --smoothingSigmas 1 --verbose 1 --output '[mat_b0groups/mat.Z0000T0000,/tmp/sct-20200517024148.039279-moco-7p9n_nbk/b0_T0000_moco.nii]' -n 'BSpline[3]' # in /tmp/sct-20200517024148.039279-moco-7p9n_nbk
terminate called after throwing an instance of 'itk::ExceptionObject'
   what():  /__w/ANTs/ANTs/antsbin/staging/include/ITK-5.1/itkImageRegistrationMethodv4.hxx:1323:
   itk::ERROR: ImageRegistrationMethodv4(0x2496420): sampling percentage outside expected (0,1] range
```

If a user makes that typo:

```
terminate called after throwing an instance of 'itk::ExceptionObject'
   what():  /__w/ANTs/ANTs/Utilities/antsCommandLineParser.h:126:
   itk::ERROR: CommandLineParser(0x29a2cc0): ERROR: Parse error occured during command line argument processing
   ERROR: Unable to convert 'none' to type 'f' as float
```

So it could hold the user's hand better but it's in line with the rest of the code.

Reply:
If you approve, I'll squash those three commits together to just keep the most recent version.

Reply:
i approve!